### PR TITLE
[WIPTEST] code cleanups in utils.appliance/db/db_queries

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
this pr cleans up `applianc.configuration_data` and simplifies `utils.db_queries`

it also introduces a `db.get_first_of(table_name, **filters)` method to the cfme db object

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit a2b4079a6b5fa102d088e78176d2f3189d1a998a:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: utils/appliance/__init__.py [hunk 1]
```diff
@@ -1617,29 +1613,23 @@ def configuration_details(self):
...
 
+    def server_id(self):
+        return self._config_detail(2)
+
     def server_region(self):
-        try:
-            return self.configuration_details[0]
-        except TypeError:
-            return None
+        return self._config_detail(0)
```
Comment (mshriver) [inline utils/appliance/__init__.py]: consider replacing these 'magic' numbers with constants.

File: utils/db_queries.py [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
-
 from utils.db import cfmedb, Db
 
 
+def _db(db=None, ip_address=None):
+    if ip_address is None:
+        ip_address = cfmedb().hostname
+
+    if db is None:
+        db = Db(hostname=ip_address)
```
Comment (mshriver) [inline utils/db_queries.py]: Consider simplifying:

```python
db = Db(ip_address or cfmedb().hostname)

File: utils/db_queries.py [hunk 3]
```diff
@@ -1,8 +1,16 @@
 # -*- coding: utf-8 -*-
-
 from utils.db import cfmedb, Db
 
 
+def _db(db=None, ip_address=None):
```
Comment (mshriver) [inline utils/db_queries.py]: great removal of the code duplication below.

File: utils/appliance/__init__.py (hunk 1 above)
Comment (RonnyPfannschmidt) [inline utils/appliance/__init__.py]: in that case i'd rather introduce namedtuples and alias properties

**Comment by dajoRH** [comment]
Lint report for commit 0de111f3a63051077e079439b73dfbf63849b541:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: utils/db_queries.py [hunk 4]
```diff
@@ -51,52 +55,26 @@ def server_filter(server):
...
-        )
-    )
-    if zones:
-        return zones[0].description
-    else:
-        return None
+    db = _db(db, ip_address)
+    zone = db.get_first_of("zones", id=zone_id)
+    if zone is not None:
+        return zone.description
```
Comment (RonnyPfannschmidt) [inline utils/db_queries.py]: thats a behavior change, lets make a extra issue?

Comment (mshriver) [inline utils/db_queries.py]: logging would be a behavior change, but before it was explicitly returning None when zone was equivalent to None. Up to you.

File: utils/appliance/__init__.py [hunk 5]
```diff
@@ -1617,29 +1613,10 @@ def configuration_details(self):
...
-            return self.configuration_details[1]
-        except TypeError:
-            return None
-
-    def server_zone_id(self):
-        try:
-            return self.configuration_details[3]
-        except TypeError:
-            return None
+    server_id = db_queries.config_detail_method(db_queries.ConfigDetailResult.server_id)
```
Comment (psav) [inline utils/appliance/__init__.py]: Are these still related to the ```self```, I'm not sure they specifically are. I have already done some cleanup in this area and did plan to tackle this at some point potentially moving much of this stuff into utils.appliance and not necessarily out of it. Thoughts??

File: utils/db_queries.py (hunk 4 above)
Comment (RonnyPfannschmidt) [inline utils/db_queries.py]: its still returning none, i think silent makes sense, @psav may know details that warrant a warning

File: utils/appliance/__init__.py (hunk 5 above)
Comment (psav) [inline utils/appliance/__init__.py]: Right, but I'm wondering if we can get rid of that whole module, ```db_queries```, the configuration details _should_ imho, be "gotten" from the configuration collection of object or something like that.

Comment (RonnyPfannschmidt) [inline utils/appliance/__init__.py]: Then let's rack that up for a card?

Comment (psav) [inline utils/appliance/__init__.py]: I guess yeh

**Comment by psav** [comment]
Let's get this tested somehow

**Comment by psav** [comment]
In PRT

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by RonnyPfannschmidt** [comment]
@psav should i retry this one or abadon?

**Comment by RonnyPfannschmidt** [comment]
abandon it is

**Pull request closed without merging**